lib/extendedkey.ex

```diff
@@ -179,34 +209,96 @@ defmodule Bitcoinex.ExtendedKey do
       :xprv -> @xprv_pfx
       :tpub -> @tpub_pfx
       :tprv -> @tprv_pfx
+      :ypub -> @ypub_pfx
+      :yprv -> @yprv_pfx
+      :upub -> @upub_pfx
+      :uprv -> @uprv_pfx
+      :zpub -> @zpub_pfx
+      :zprv -> @zprv_pfx
+      :vpub -> @vpub_pfx
+      :vprv -> @vprv_pfx
     end
   end
 
+  defp bip44 do
+    [
+      @xpub_pfx,
+      @xprv_pfx,
+      @tpub_pfx,
+      @tprv_pfx
+    ]
+  end
+
+  defp bip49 do
+    [
+      @ypub_pfx,
+      @yprv_pfx,
+      @upub_pfx,
+      @uprv_pfx
+    ]
+  end
+
+  defp bip84 do
+    [
+      @zpub_pfx,
+      @zprv_pfx,
+      @vpub_pfx,
+      @vprv_pfx
+    ]
+  end
+
   defp prv_to_pub_prefix(prv_pfx) do
     case prv_pfx do
       @xprv_pfx -> @xpub_pfx
       @tprv_pfx -> @tpub_pfx
+      @yprv_pfx -> @ypub_pfx
+      @uprv_pfx -> @upub_pfx
+      @zprv_pfx -> @zpub_pfx
+      @vprv_pfx -> @vpub_pfx
     end
   end
 
   defp mainnet_prefixes do
     [
       @xpub_pfx,
-      @xprv_pfx
+      @xprv_pfx,
+      @ypub_pfx,
+      @yprv_pfx,
+      @zpub_pfx,
+      @zprv_pfx
     ]
   end
 
   @spec network_from_prefix(binary) :: atom
-  defp network_from_prefix(prefix) do
+  def network_from_prefix(prefix) do
     if prefix in mainnet_prefixes(), do: :mainnet, else: :testnet
   end
 
-  @doc """
-    network_from_extended_key returns :testnet or :mainnet 
-    depending on the network prefix of the key.
-  """
-  @spec network_from_extended_key(t()) :: atom
-  def network_from_extended_key(%__MODULE__{prefix: prefix}), do: network_from_prefix(prefix)
+  @spec script_type_from_prefix(binary) :: atom
+  def script_type_from_prefix(prefix) do
+    cond do
+      prefix in bip44() -> :p2pkh
+      # p2sh or p2sh_p2wpkh?
+      prefix in bip49() -> :p2sh_p2wpkh
+      prefix in bip84() -> :p2wpkh
+    end
+  end
+
+  @spec switch_prefix(t(), atom) :: t() | {:error, String.t()}
```

Comment:
I think it's more consistent to wrap succes type in :ok tag like `{:ok, t()}` if  error tuple is another possible return value